software/Makefile

```diff
@@ -99,10 +99,12 @@ clean_logs:
 
 # run on x86
 VPATH := $(wildcard ../cosim/import/black-parrot-sdk/prog/*)
+# FPGAs should not skip zeros
```

Comment:
@farzamgl what do you think is the best way to handle this?

Reply:
maybe in PS.cpp we can just zero the DRAM explicitly rather than relying on NBF to do it? That seems cleaner

Reply:
I don't think we need to zero out the entire memory. We can just keep the zeros that are specified in the .mem file(sections like .data etc) and not add extra zeros for the rest of the memory space. This is what I do right now and I don't have issues with the FPGA.

Reply:
Basically undoing this change: https://github.com/black-parrot/black-parrot/commit/87e6207cb691234a9d1677647d3d14e468a37408#diff-3392d47b697b464c442cac328e5de48b1605b3667ac7c84a06eef4e4987454dbR208

However, I think there are some scenarios where we have to zero the entire memory because of how the binary is built. For example, we include Linux as a payload to OpenSBI, so we lose the section labels for the privileged code and we need to zero out everything else to be safe.

Reply:
How long does it take to zero the whole memory vs a program?

Reply:
The latter takes significantly less time because for example the .data section is a small portion of the entire memory space.

Reply:
Right, but are we talking 1s or 1m or what order of magnitude? I think it's fine to be conservative if it stops users from having issues and doesn't impact the overall throughput too badly?

Reply:
Haven't tried zeroing the entire memory in PS before so don't know the time. Shouldn't be more than a couple of minutes. I don't know how I feel about making it the default behavior because it's not going to be necessary for the vast majority of cases where users are just running bare-metal code.

Reply:
Makes sense, I'll add a #define for zero-ing the DRAM from PS